src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSpec.scala

```diff
@@ -872,6 +879,68 @@ class ClusterMonitorSpec extends TestKit(ActorSystem("leonardotest")) with FlatS
     }
   }
 
+  // Pre:
+  // - cluster exists in the DB with status Starting
+  // - dataproc DAO returns status RUNNING
+  // - Jupyter DAO returns status as False
+  // - compute DAO returns all instances running
+  // Post:
+  // - cluster is not updated in the DB with status Running
+  it should "cluster should not got from STARTING to RUNNING if Jupyter is not ready" in isolatedDbTest {
+    dbFutureValue { _.clusterQuery.save(startingCluster, gcsPath("gs://bucket"), Some(serviceAccountKey.id)) } shouldEqual startingCluster
+
+    val gdDAO = mock[GoogleDataprocDAO]
+    when {
+      gdDAO.getClusterStatus(mockitoEq(startingCluster.googleProject), mockitoEq(startingCluster.clusterName))
+    } thenReturn Future.successful(ClusterStatus.Running)
+    when {
+      gdDAO.getClusterInstances(mockitoEq(startingCluster.googleProject), mockitoEq(startingCluster.clusterName))
+    } thenReturn Future.successful(clusterInstances)
+    when {
+      gdDAO.getClusterMasterInstance(mockitoEq(startingCluster.googleProject), mockitoEq(startingCluster.clusterName))
+    } thenReturn Future.successful(Some(masterInstance.key))
+    when {
+      gdDAO.getClusterStagingBucket(mockitoEq(startingCluster.googleProject), mockitoEq(startingCluster.clusterName))
+    } thenReturn Future.successful(Some(GcsBucketName("staging-bucket")))
+
+    val computeDAO = stubComputeDAO(InstanceStatus.Running)
+    val storageDAO = mock[GoogleStorageDAO]
+    when {
+      storageDAO.deleteBucket(any[GcsBucketName], any[Boolean])
+    } thenReturn Future.successful(())
+
+    when {
+      storageDAO.setBucketAccessControl(any[GcsBucketName], any[GcsEntity], any[GcsRole])
+    } thenReturn Future.successful(())
+
+    when {
+      storageDAO.setDefaultObjectAccessControl(any[GcsBucketName], any[GcsEntity], any[GcsRole])
+    } thenReturn Future.successful(())
+
+    val iamDAO = mock[GoogleIamDAO]
+    when {
+      iamDAO.removeIamRolesForUser(any[GoogleProject], any[WorkbenchEmail], mockitoEq(Set("roles/dataproc.worker")))
+    } thenReturn Future.successful(())
+    val authProvider = mock[LeoAuthProvider]
+
+
+    val jupyterDAO = mock[JupyterDAO]
+    when {
+      jupyterDAO.getStatus(mockitoEq(startingCluster.googleProject), mockitoEq(startingCluster.clusterName))
+    } thenReturn Future.successful(false)
+
+    withClusterSupervisor(gdDAO, computeDAO, iamDAO, storageDAO, authProvider, jupyterDAO, false) { actor =>
+      actor ! ClusterCreated(startingCluster)
+      expectNoMessage(1 second)
+
+      val updatedCluster = dbFutureValue { _.clusterQuery.getActiveClusterByName(startingCluster.googleProject, startingCluster.clusterName) }
+      updatedCluster shouldBe 'defined
+      updatedCluster.map(_.status) shouldBe Some(ClusterStatus.Starting)
+      actor ! PoisonPill
```

Comment:
Subsequent tests fail if the actor is not shutdown. not sure why

Reply:
Interesting that this is needed. I think `withClusterSupervisor` is supposed to handle actor teardown...